classes/actions/ValidationOrderActions.php

```diff
@@ -813,15 +813,8 @@ public function chargeWebhook()
                 break;
 
             case 'charge.refunded':
-                if ($order->getCurrentState() != Configuration::get('PS_OS_PAYMENT')) {
-                    $order->setCurrentState(Configuration::get('PS_OS_CANCELED'));
-                    ProcessLoggerHandler::logInfo(
-                        'Order canceled',
-                        null,
-                        null,
-                        'ValidationOrderActions - chargeWebhook'
-                    );
-                } else {
+                $currentState = new \OrderState($order->getCurrentState());
+                if ($currentState->paid === true) {
```

Comment:
With older PrestaShop version, if you make a strict equal you should cast the value:
(bool) $currentState->paid

Reply:
Done !